controllers/watcher_controller.go

```diff
@@ -116,11 +116,52 @@ type EndpointDetail struct {
 func (ed *EndpointDetail) ensureRoute(
 	ctx context.Context,
 	helper *helper.Helper,
+	instance *watcherv1beta1.Watcher,
+	svc *k8s_corev1.Service,
 ) (ctrl.Result, error) {
+	// check if there is already a route with watcherapi labels
+	routes, err := GetRoutesListWithLabel(ctx, helper, instance.Namespace, getAPIServiceLabels())
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	for _, route := range routes.Items {
+		instanceRef := metav1.OwnerReference{
+			APIVersion:         instance.APIVersion,
+			Kind:               instance.Kind,
+			Name:               instance.GetName(),
+			UID:                instance.GetUID(),
+			BlockOwnerDeletion: ptr.To(true),
+			Controller:         ptr.To(true),
+		}
+
+		owner := metav1.GetControllerOf(&route.ObjectMeta)
+
+		// Delete the route if the service was changed not to expose a route
+		if svc.ObjectMeta.Annotations[service.AnnotationIngressCreateKey] == "false" &&
+			route.Spec.To.Name == ed.Name &&
+			owner != nil && owner.UID == instance.GetUID() {
+			// Delete any other owner refs from ref list to not block deletion until owners are gone
```

Comment:
that does not sound like the correct behavior however i think we can address this later.
this will be racy if those refs were added by another operator.